middle_end/flambda/simplify/env/upwards_env.ml

```diff
@@ -80,9 +80,11 @@ let resolve_exn_continuation_aliases t exn_cont =
 
 let continuation_arity t cont =
   match find_continuation t cont with
-  | Other { arity; handler = _; }
-  | Unreachable { arity; }
-  | Linearly_used_and_inlinable { arity; _ } -> arity
+  | Linearly_used_and_inlinable { params; _ }
+  | Non_inlinable { params; _ } ->
+    Kinded_parameter.List.arity_with_subkinds params
+  | Toplevel_or_function_return_or_exn_continuation { arity; }
+  | Unreachable { arity; } -> arity
```

Comment:
Couldn't the `Continuation_in_env.arity` function be used here instead of re-implementing it ?

Reply:
Yes indeed.